src/main/java/com/fivetran/external/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -552,6 +557,42 @@ private boolean causedByStreamRecordProcessingError(Throwable t) {
         return causedByStreamRecordProcessingError(t.getCause());
     }
 
+    private void handleTrimmedDataAccessException(Exception e) {
+        Optional<TrimmedDataAccessException> maybeTrimmedException = getTrimmedDataAccessException(e);
+        if (maybeTrimmedException.isPresent()) throw maybeTrimmedException.get();
+    }
+
+    private Optional<TrimmedDataAccessException> getTrimmedDataAccessException(Throwable t) {
```

Comment:
nit: It looks like we use this recursive way to get exception that we care about here, either `TrimmedDataAccessException` or `MissingIncompleteLeasesException`, but it looks like they are more or less doing the same thing, just checking for a different exception class. Could we parameterize the exception class and just use one method here?

Reply:
Nice call. I will fix this.